### PR TITLE
Add option to disable hotkeys

### PR DESCRIPTION
**Pull request by belcherj**
### Fix

Adds setting to disable hotkeys

### Test

1. Load page
2. Try hotkeys, they should work
3. Disable in settings
4. They should not work

Screen Shot 2020-05-13 at 10 10 49 PM

### CODE REVIEW

File: lib/state/settings/actions.ts [hunk 1]
```diff
@@ -49,6 +49,15 @@ export const setLineLength: A.ActionCreator<A.SetLineLength> = (
   lineLength,
 });
 
+export const toggleDisableHotkeys = () => (dispatch, getState) => {
+  const newValue = !getState().settings.disableHotkeys;
```
Comment (codebykat) [inline lib/state/settings/actions.ts]: For simplicity and consistency with the rest of the actions, I don't think we need to set a variable here, we can just pass the value below like `sortReversed` etc. is doing.

**Review by codebykat — CHANGES_REQUESTED** [review]
Okay I have 2.5 comments about this that I feel pretty strongly about:

Screen Shot 2020-05-13 at 3 18 29 AM

1. Can we not call them "hotkeys"? They're already "Keyboard shortcuts" in the popup dialog and that sounds more friendly / obvious.

2. Right now this is an "Enable to disable" toggle which personally I find annoying at best, and kind of malicious at worst. Can we make this setting "Enable keyboard shortcuts" and have it default to on? Same functionality, much less confusing.

For comparison, Fastmail does this, very classy:

Screen Shot 2020-05-13 at 3 20 32 AM

And Gmail does this:

Screen Shot 2020-05-13 at 3 21 01 AM

The last half-a-comment is, should this really be under Display? There's hella room on Tools (whereas Display has a ton of stuff in it already) and IMO it's marginally more logical there.

All of that aside, the code and functionality are okay.

**Comment by SylvesterWilmott** [comment]
Agree with @codebykat 

1. Let's call them "Keyboard Shortcuts" everywhere
2. Default to on and have the toggle just be labelled "Keyboard Shortcuts"

**Comment by SylvesterWilmott** [comment]
>The last half-a-comment is, should this really be under Display? There's hella room on Tools (whereas Display has a ton of stuff in it already) and IMO it's marginally more logical there.

Tools makes a lot more sense to me

File: lib/tag-field/index.tsx [hunk 2]
```diff
@@ -282,6 +293,12 @@ export class TagField extends Component<Props, OwnState> {
...
->)(TagField);
+const mapStateToProps: S.MapState<StateProps> = ({
+  settings: { keyboardShortcuts },
+}) => ({
+  keyboardShortcuts,
+});
+
+export default connect(mapStateToProps, {
+  updateNoteTags,
+} as S.MapDispatch<DispatchProps>)(TagField);
```
Comment (codebykat) [inline lib/tag-field/index.tsx:304]: Fun fact: this is the only place in the codebase we're doing this (as opposed to defining a separate `MapDispatchToProps` variable) and I don't think I like it. I do realize it was already like that, though.

**Review by codebykat — APPROVED** [review]
One minor code comment, but this looks beautiful and works like a dream now. :shipit:

**Comment by SylvesterWilmott** [comment]
Design looks good! Thanks @belcherj

File: lib/tag-field/index.tsx (hunk 2 above)
Comment (belcherj) [reply lib/tag-field/index.tsx:304]: I'll do a follow up

**Pull request merged**